waspc/src/Wasp/Project/Analyze.hs

```diff
@@ -1,32 +1,45 @@
 module Wasp.Project.Analyze
   ( analyzeWaspProject,
-    readPackageJsonFile,
     analyzeWaspFileContent,
     findWaspFile,
     findPackageJsonFile,
     analyzePrismaSchema,
+    WaspFile (..),
   )
 where
 
+import Control.Applicative ((<|>))
 import Control.Arrow (ArrowChoice (left))
+import Control.Concurrent (newChan)
+import Control.Concurrent.Async (concurrently)
+import Control.Monad.Except (ExceptT (..), runExceptT)
 import qualified Data.Aeson as Aeson
+import Data.Conduit.Process.Typed (ExitCode (..))
 import Data.List (find, isSuffixOf)
-import StrongPath (Abs, Dir, File', Path', toFilePath, (</>))
+import StrongPath (Abs, Dir, File', Path', Rel, basename, toFilePath, (</>))
+import qualified StrongPath as SP
+import StrongPath.TH (relfile)
+import StrongPath.Types (File)
```

Comment:
[Heh](https://i.giphy.com/l0HU8V1CHKTUFtuFO.webp).

I have had this item on my Haskell list for quite some time.

For some reason (and for some SP imports), LSP _almost never_ suggests a top-level import from `StrongPath`. Whenever I look at LSP's import suggestions, this is my protocol:
1. Check if there is an `Add to import` option.
2. Check if there is a top-level import option (in this case, `StrongPath`).
3. Use a lower-level import option and fix the import manually later (sometimes it works, sometimes it doesn't).

This mental overhead drives me crazy, and sometimes I forget to change the imports manually later (like here). I would _really_ like to solve this because I don't want to waste time manually editing the imports.

@Martinsos @infomiho Is this specific to my setup, does it happen to you?
Please try with `reldirP` in `Wasp.Project.Common`. I just reproduced it there:

![image](https://github.com/user-attachments/assets/cf0fd6d2-7f5d-4c0f-85db-ad22701649cb)

Reply:
<img width="686" alt="Screenshot 2024-09-30 at 13 13 41" src="https://github.com/user-attachments/assets/23914ce4-c666-4c0f-8ea3-17c9de89201e">

Top level for me

Reply:
I am in emacs, so it is a bit different, but first I get a suggestion for autocompletion:

![image](https://github.com/user-attachments/assets/756e07e2-d6ad-4269-8190-ca1081bb9297)

This one is correct, but when I accept, for some reason it doesn't also add an import automatically. So I get a compiler error that symbol is missing. I then tell LSP to suggest code actions, and then I get this

![image](https://github.com/user-attachments/assets/4c9cb4ff-dea7-443a-ab3b-af99816991c2)

Which is correct! I am offered also the lower level imports, but the first option I am offered is to add it to existing import list of StrongPath, and if I pick that, it adds import correctly.

Reply:
Ok, glad to hear it's fixable because it's been annoying me for ages (particularly with StrongPath imports). I'll bother you in the office to help me figure it out.